samples/go/csv/read.go

```diff
@@ -113,7 +113,13 @@ func ReadToList(r *csv.Reader, structName string, headers []string, kinds KindSl
 		}
 
 		fields := readFieldsFromRow(row, headers, fieldOrder, kindMap)
-		valueChan <- types.NewStructWithType(t, fields)
+		data := make(types.StructData, len(fields))
```

Comment:
this is going to regress csv reader perf. can you open a bug to new `NewStructWithFields`?